infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -285,12 +285,35 @@ private void rangeWrite(CorfuPayloadMsg<RangeWriteMsg> msg,
         r.sendResponse(ctx, msg, CorfuMsgType.WRITE_OK.msg());
     }
 
+    /**
+     * Seal the server with the epoch by setting epochWaterMark.
+     *
+     * - The epochWaterMark is persisted to withstand restarts.
+     * - The epochWaterMark is inserted in the queue and then we wait to flush all operations
+     *   in the queue before this operation.
+     * - All operations after this operation but stamped with an older epoch will be failed.
+     */
+    @Override
+    public synchronized void sealServerWithEpoch(long epoch, IServerRouter r) {
+        if (epoch > serverContext.getLogUnitEpochWaterMark()) {
```

Comment:
Just pass the epoch the router is not needed. Only the base server needs to know about the router (i.e. set in mem epoch)

Reply:
Done, I forgot to remove it.